File: Data Structures & Algorithms/two-integer-sum-ii/submission-2.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int frontPointer = 0;
        int backPointer = numbers.length -1; 

        int[] ans = new int[2];

        while (frontPointer < backPointer){
            int frontNumber = numbers[frontPointer];
            int backNumber = numbers[backPointer];
            if (frontNumber + backNumber == target){
                ans[0] = frontPointer + 1;
                ans[1] = backPointer + 1;
                return ans;
                
            }
            if (frontNumber + backNumber > target){
                backPointer -=1;
            }
            else{
                frontPointer += 1;
            }
        }
        return(ans);        
    }
}

File: Data Structures & Algorithms/is-palindrome/submission-2.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        frontpointer = 0
        backpointer = len(s) - 1

        while(frontpointer < backpointer):
            if(not s[frontpointer].isalnum() ):
                frontpointer += 1
            elif(not s[backpointer].isalnum()):
                backpointer -= 1
            elif(s[frontpointer].lower() != s[backpointer].lower()):
                return False
            else:
                frontpointer += 1
                backpointer -= 1
        
        return True